numba/tests/test_jit_module.py

```diff
@@ -0,0 +1,155 @@
+
+import os
+import sys
+import shutil
+import inspect
+import importlib
+import contextlib
+import uuid
+import numpy as np
+import logging
+
+import numba.unittest_support as unittest
+from numba import dispatcher
+from numba.utils import StringIO
+from numba.tests.support import temp_directory
+
+logger = logging.getLogger('numba.decorators')
+
+
+class CapturedLog:
+    """Capture the log temporarily for validation."""
+
+    def __init__(self):
+        self.buffer = StringIO()
+        self.handler = logging.StreamHandler(self.buffer)
+
+    def __enter__(self):
+        self._handlers = logger.handlers
+        self.buffer = StringIO()
+        logger.handlers = [logging.StreamHandler(self.buffer)]
+
+    def __exit__(self, type, value, traceback):
+        logger.handlers = self._handlers
+
+    def getvalue(self):
+        return self.buffer.getvalue()
+
+
+class TestJitModule(unittest.TestCase):
+
```

Comment:
Given these tests are leaving state on the file system, they probably need to inherit from `numba.support.SerialMixin`

Reply:
no problem. thanks for adding it.